newIDE/electron-app/app/main.js

```diff
@@ -190,6 +190,31 @@ app.on('ready', function() {
     );
   });
 
+  // Yarn Dialogue Tree Editor
+  ipcMain.on('yarn-create-json', (event, externalEditorData) => {
+    loadModalWindow({
+      parentWindow: mainWindow,
+      devTools,
+      readyChannelName: 'yarn-ready',
+      indexSubPath: 'yarn/yarn-index.html',
+      relativeWidth: 0.55,
+      relativeHeight: 0.8,
+      backgroundColor: '#000000',
+      onReady: yarnWindow => {
+        yarnWindow.webContents.send('yarn-open', externalEditorData);
+        yarnWindow.show();
+      },
+    });
+  });
+
+  ipcMain.on('yarn-changes-saved', (event, newFilePath, externalEditorData) => {
```

Comment:
Seems like externalEditorData is not used, remove it from the `.on` and the `send`